diff -pruN portage-utils-0.63.orig/main.c portage-utils-0.63/main.c
--- portage-utils-0.63.orig/main.c	2016-03-28 05:53:50.000000000 +0200
+++ portage-utils-0.63/main.c	2016-03-29 02:40:11.935979233 +0200
@@ -1422,7 +1422,8 @@ int main(int argc, char **argv)
 
 	if (fstat(fileno(stdout), &st) != -1)
 		if (!isatty(fileno(stdout)))
-			no_colors();
+			if (S_ISFIFO(st.st_mode) == 0)
+				no_colors();
 	if ((getenv("TERM") == NULL) || (strcmp(getenv("TERM"), "dumb") == 0))
 		no_colors();
 
diff -pruN portage-utils-0.63.orig/qlop.c portage-utils-0.63/qlop.c
--- portage-utils-0.63.orig/qlop.c	2016-03-28 05:53:50.000000000 +0200
+++ portage-utils-0.63/qlop.c	2016-03-29 02:42:29.418960636 +0200
@@ -32,7 +32,7 @@ static const char * const qlop_opts_help
 	"Show sync history",
 	"Show current emerging packages",
 	"Limit selection to this time (1st -d is start, 2nd -d is end)",
-	"Read emerge logfile instead of $EMERGE_LOG_DIR/" QLOP_DEFAULT_LOGFILE,
+	"Read emerge logfile instead of $EMERGE_LOG_DIR/" QLOP_DEFAULT_LOGFILE "\n\t\t\t (use \"-\" for stdin)",
 	COMMON_OPTS_HELP
 };
 static const char qlop_desc[] =
@@ -47,6 +47,30 @@ static const char qlop_desc[] =
 #define QLOP_LIST    0x01
 #define QLOP_UNLIST  0x02
 
+
+static FILE *
+qfopen(const char *logfile)
+{
+	FILE *fp;
+	if (strcmp(logfile, "-") == 0)
+		return stdin;
+	if ((fp = fopen(logfile, "r")))
+		return fp;
+	else {
+		warnp("Could not open logfile '%s'", logfile);
+		exit(EXIT_FAILURE);
+	}
+}
+
+static void
+qfclose(FILE *fp)
+{
+	if (fp != stdin)
+		fclose(fp);
+	else
+		exit(EXIT_SUCCESS);
+}
+
 static void
 print_seconds_for_earthlings(const unsigned long t)
 {
@@ -98,10 +122,7 @@ show_merge_times(char *package, const ch
 		pkg = package;
 	}
 
-	if ((fp = fopen(logfile, "r")) == NULL) {
-		warnp("Could not open logfile '%s'", logfile);
-		return 1;
-	}
+	fp = qfopen(logfile);
 
 	while (fgets(buf[0], sizeof(buf[0]), fp) != NULL) {
 		if (strstr(buf[0], pkg) == NULL)
@@ -211,7 +232,7 @@ show_merge_times(char *package, const ch
 			atom_implode(atom);
 		}
 	}
-	fclose(fp);
+	qfclose(fp);
 	if (count == 0)
 		return 0;
 	if (average == 1) {
@@ -240,10 +261,7 @@ show_emerge_history(int listflag, array_
 	time_t t;
 	depend_atom *atom, *logatom;
 
-	if ((fp = fopen(logfile, "r")) == NULL) {
-		warnp("Could not open logfile '%s'", logfile);
-		return;
-	}
+	fp = qfopen(logfile);
 
 	buf = NULL;
 	while ((linelen = getline(&buf, &buflen, fp)) != -1) {
@@ -305,7 +323,7 @@ show_emerge_history(int listflag, array_
 	}
 
 	free(buf);
-	fclose(fp);
+	qfclose(fp);
 }
 
 /* The format of the sync log has changed over time.
@@ -335,10 +353,7 @@ show_sync_history(const char *logfile, t
 	char *buf, *p;
 	time_t t;
 
-	if ((fp = fopen(logfile, "r")) == NULL) {
-		warnp("Could not open logfile '%s'", logfile);
-		return;
-	}
+	fp = qfopen(logfile);
 
 	buf = NULL;
 	/* Just find the finish lines. */
@@ -373,7 +388,7 @@ show_sync_history(const char *logfile, t
 	}
 
 	free(buf);
-	fclose(fp);
+	qfclose(fp);
 }
 
 static void show_current_emerge(void);
