--- mkchromecast-0.3.8.1/setup.py.orig	2017-12-24 15:27:52.000000000 +0100
+++ mkchromecast-0.3.8.1/setup.py	2020-03-06 10:27:12.443707418 +0100
@@ -16,9 +16,16 @@
     google
 """
 from setuptools import setup
+import os
 
 version=open('mkchromecast/version.py').readlines()[-1].split()[-1].strip("\"'")
 
+def get_data_files(folder):
+    data_files = []
+    for root, dir, files in os.walk(folder):
+        data_files.append((root, [f"{root}/{f}" for f in files]))
+    return data_files
+
 APP = ['start_tray.py']
 APP_NAME = 'Mkchromecast'
 DATA_FILES = [
@@ -32,9 +39,7 @@
         'images/google_working_w.icns',
         'images/google_nodev_w.icns',
         'bin/audiodevice',
-        'nodejs',
-        'notifier'
-        ]
+        ] + get_data_files('nodejs') + get_data_files('notifier')
 
 OPTIONS = {
     'argv_emulation': True,
@@ -64,6 +69,7 @@
     }
 }
 
+import os
 setup(
     name=APP_NAME,
     app=APP,
