--- a/lib/authlogic_openid/acts_as_authentic.rb.~1~	2011-07-17 21:45:44.000000000 +0200
+++ b/lib/authlogic_openid/acts_as_authentic.rb	2018-06-29 14:05:47.350679614 +0200
@@ -81,7 +81,6 @@
       def save(options={}, &block)
         return false if options[:validate] != false && block_given? && authenticate_with_openid? && !authenticate_with_openid
         result = super
-        yield(result) if block_given?
         result
       end
 
