--- epiphany-0.7.0/src/game_ori.cpp	2012-04-27 10:07:36.090005197 +0200
+++ epiphany-0.7.0/src/game.cpp	2012-04-27 11:13:12.790004284 +0200
@@ -325,9 +325,7 @@
 	
 	char text[255];
 	
-	sprintf(text, "%s",("Score: "));
-	
-	sprintf(text, "%s%d", text,(Sint32)m_level->get_player().get_score());
+	sprintf(text, "Score: %d", (Sint32)m_level->get_player().get_score());
 	
 	game_font->write(4,real_game_size_y+5, text);
 
@@ -336,8 +334,7 @@
 
 	if(remaining>0)
 	{
-		sprintf(text,"%s","Remaining:   ");
-		sprintf(text,"%s%d", text, remaining);
+		sprintf(text,"Remaining:   %d", remaining);
 		game_font->write(200,real_game_size_y+5, text);
 
 	}
@@ -414,14 +411,10 @@
   		
 			char current_level_path[255];
   		
-			sprintf(current_level_path, "%s", (Resource_Factory::instance()->get_resource_path().c_str()));
-  		
-			sprintf(current_level_path, "%s%s", current_level_path, "/maps/level");
-  	
-			sprintf(current_level_path, "%s%d", current_level_path, (menu.get_current_level()));
-  		
-			sprintf(current_level_path, "%s%s", current_level_path, ".map");
-  	
+			sprintf(current_level_path, "%s/maps/level%d.map",
+				(Resource_Factory::instance()->get_resource_path().c_str()),
+				(menu.get_current_level()));
+
 			DEBOUT("Loading map: "<<current_level_path<<"\n");
   	
 			m_level->load_map(current_level_path);
