--- lib/rvideo/tools/ffmpeg.rb.orig	2012-03-20 10:06:21.000000000 +0100
+++ lib/rvideo/tools/ffmpeg.rb	2012-03-20 09:57:31.000000000 +0100
@@ -59,7 +59,7 @@
           raise TranscoderError, "Could not write output file to #{@output_file}"
         end
           
-        full_details = /Press .* to stop encoding\n(.*)/m.match(result)
+        full_details = /Press .* to stop.*?\n(.*)/m.match(result)
         raise TranscoderError, "Unexpected result details (#{result})" if full_details.nil?
         details = full_details[1].strip.gsub(/\s*\n\s*/," - ")
         
