--- lib/system_check/rake_task/app_task.rb.orig	2019-01-06 14:52:33.019818978 +0100
+++ lib/system_check/rake_task/app_task.rb	2019-01-06 14:52:47.103819703 +0100
@@ -23,8 +23,6 @@
           SystemCheck::App::UploadsDirectoryExistsCheck,
           SystemCheck::App::UploadsPathPermissionCheck,
           SystemCheck::App::UploadsPathTmpPermissionCheck,
-          SystemCheck::App::InitScriptExistsCheck,
-          SystemCheck::App::InitScriptUpToDateCheck,
           SystemCheck::App::ProjectsHaveNamespaceCheck,
           SystemCheck::App::RedisVersionCheck,
           SystemCheck::App::RubyVersionCheck,
