diff --git a/SortFilterProxyModel.pro b/SortFilterProxyModel.pro
index 57e3175..dea9731 100644
--- a/SortFilterProxyModel.pro
+++ b/SortFilterProxyModel.pro
@@ -16,6 +16,10 @@ unix:!mac:isEmpty(PREFIX) {
     PREFIX = /usr/local
 }
 
+isEmpty(LIB_DIR){
+    LIB_DIR = $${PREFIX}/lib
+}
+
 headers.path = $${PREFIX}/include/SortFilterProxyModel
 headers.files = $$PWD/qqmlsortfilterproxymodel.h
 
@@ -50,7 +54,7 @@ proxyrole_headers.files = $$PWD/proxyroles/proxyrole.h \
     $$PWD/proxyroles/regexprole.h \
     $$PWD/proxyroles/filterrole.h
 
-target.path = $${PREFIX}/lib
+target.path = $${LIB_DIR}
 
 QMAKE_PKGCONFIG_NAME = SortFilterProxyModel
 QMAKE_PKGCONFIG_DESCRIPTION = A nicely exposed QSortFilterProxyModel for QML
