diff -Npur moria-5.6.orig/source/files.c moria-5.6/source/files.c
--- moria-5.6.orig/source/files.c	2011-11-05 12:21:31.823290449 +0100
+++ moria-5.6/source/files.c	2011-11-05 12:44:34.703290350 +0100
@@ -126,19 +126,19 @@ void read_times()
 	if (strlen(in_line) > 3)
 	  {
 	    if (!strncmp(in_line, "SUN:", 4))
-	      (void) strcpy(days[0], in_line);
+	      (void) snprintf(days[0], sizeof(days[0]), "%s", in_line);
 	    else if (!strncmp(in_line, "MON:", 4))
-	      (void) strcpy(days[1], in_line);
+	      (void) snprintf(days[1], sizeof(days[0]), "%s", in_line);
 	    else if (!strncmp(in_line, "TUE:", 4))
-	      (void) strcpy(days[2], in_line);
+	      (void) snprintf(days[2], sizeof(days[0]), "%s", in_line);
 	    else if (!strncmp(in_line, "WED:", 4))
-	      (void) strcpy(days[3], in_line);
+	      (void) snprintf(days[3], sizeof(days[0]), "%s", in_line);
 	    else if (!strncmp(in_line, "THU:", 4))
-	      (void) strcpy(days[4], in_line);
+	      (void) snprintf(days[4], sizeof(days[0]), "%s", in_line);
 	    else if (!strncmp(in_line, "FRI:", 4))
-	      (void) strcpy(days[5], in_line);
+	      (void) snprintf(days[5], sizeof(days[0]), "%s", in_line);
 	    else if (!strncmp(in_line, "SAT:", 4))
-	      (void) strcpy(days[6], in_line);
+	      (void) snprintf(days[6], sizeof(days[0]), "%s", in_line);
 	  }
       (void) fclose(file1);
     }
