diff -aruN a/mariadb_config/mariadb_config.c.in b/mariadb_config/mariadb_config.c.in
--- a/mariadb_config/mariadb_config.c.in	2015-01-29 13:21:48.000000000 -0500
+++ b/mariadb_config/mariadb_config.c.in	2015-01-30 13:03:01.312548413 -0500
@@ -6,7 +6,7 @@
 #define INCLUDE "-I@PREFIX_INSTALL_DIR@/@INCLUDE_INSTALL_DIR@/@SUFFIX_INSTALL_DIR@ -I@PREFIX_INSTALL_DIR@/@INCLUDE_INSTALL_DIR@/@SUFFIX_INSTALL_DIR@/mysql"
 #define LIBS    "-L@PREFIX_INSTALL_DIR@/@LIB_INSTALL_DIR@/@SUFFIX_INSTALL_DIR@ -lmariadb" \
                 "@extra_dynamic_LDFLAGS@"
-#define CFLAGS  INCLUDE " @CMAKE_C_FLAGS@"
+#define CFLAGS  INCLUDE
 #define VERSION "@MYSQL_CLIENT_VERSION@"
 #define PLUGIN_DIR "@PREFIX_INSTALL_DIR@/@PLUGIN_INSTALL_DIR@"
 #define SOCKET  "@MYSQL_UNIX_ADDR@"
