diff -aurN a/cmake/install.cmake b/cmake/install.cmake
--- a/cmake/install.cmake	2015-12-08 02:50:35.000000000 -0500
+++ b/cmake/install.cmake	2016-02-12 21:40:46.599011426 -0500
@@ -120,12 +120,6 @@
   SET(PREFIX_INSTALL_DIR_${INSTALL_LAYOUT} ${PREFIX_INSTALL_DIR})
 ENDIF()
 
-IF(NOT SUFFIX_INSTALL_DIR)
-  SET(SUFFIX_INSTALL_DIR_${INSTALL_LAYOUT} "mariadb")
-ELSE()
-  SET(SUFFIX_INSTALL_DIR_${INSTALL_LAYOUT} ${SUFFIX_INSTALL_DIR})
-ENDIF()
-
 FOREACH(dir "BIN" "LIB" "INCLUDE" "DOCS" "PREFIX" "SUFFIX" "PLUGIN")
   SET(${dir}_INSTALL_DIR ${${dir}_INSTALL_DIR_${INSTALL_LAYOUT}})
   MARK_AS_ADVANCED(${dir}_INSTALL_DIR)
--- a/mariadb_config/CMakeLists.txt	2016-02-12 21:21:06.981021227 -0500
+++ b/mariadb_config/CMakeLists.txt	2016-02-12 22:06:02.397130410 -0500
@@ -33,4 +33,4 @@
 # Installation
 #
 INSTALL(TARGETS mariadb_config
-        DESTINATION "bin")
+        DESTINATION ${BIN_INSTALL_DIR})
