diff -aruN a/CMakeLists.txt b/CMakeLists.txt
--- a/libmysql/CMakeLists.txt	2013-05-17 14:16:47.621519747 -0400
+++ b/libmysql/CMakeLists.txt	2013-05-17 14:18:04.570428917 -0400
@@ -166,9 +166,9 @@
 
 INSTALL(TARGETS
           libmariadb mariadbclient
-          RUNTIME DESTINATION "lib"
-          LIBRARY DESTINATION "lib"
-          ARCHIVE DESTINATION "lib")
+          RUNTIME DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+          LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+          ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
 INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/include
         DESTINATION mariadbclient)
