diff --git a/configure.ac b/configure.ac
index 220ed80d..9dfe8b5f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -46,7 +46,7 @@ AM_CONDITIONAL([HAVE_GUI], [test "x$with_gui" = "xyes"])
    
 AC_PROG_CC
 AC_PROG_CXX
-AM_CONDITIONAL([HAVE_GCC], [test $CXX = g++])
+AM_CONDITIONAL([HAVE_GCC], [test $CXX = $CXX])
 AC_CHECK_SIZEOF([long unsigned int])
 AM_CONDITIONAL([HAVE_64BIT], [test $ac_cv_sizeof_long_unsigned_int = 8])
 
diff --git a/cppForSwig/cryptopp/configure.ac b/cppForSwig/cryptopp/configure.ac
index 8e944943..7583c16f 100644
--- a/cppForSwig/cryptopp/configure.ac
+++ b/cppForSwig/cryptopp/configure.ac
@@ -86,7 +86,7 @@ AC_MSG_RESULT([$CLANG])
 AC_SUBST([CLANG])
 
 AM_CONDITIONAL([CRYPTOPP_HAVE_CLANG], [test x$CLANG = xyes])
-AM_CONDITIONAL([CRYPTOPP_HAVE_GCC], [test $CXX == g++])
+AM_CONDITIONAL([CRYPTOPP_HAVE_GCC], [test $CXX == $CXX])
 
 #set this otherwise autoconf complains about conditionals never getting defined
 AM_CONDITIONAL([GAS210_OR_LATER], [test "a" == "b"])
