diff --git a/cppForSwig/Makefile b/cppForSwig/Makefile
index ba6cc45..c07d7f8 100755
--- a/cppForSwig/Makefile
+++ b/cppForSwig/Makefile
@@ -1,8 +1,6 @@
 COMPILER = g++ 
 #COMPILER_OPTS = -c -g -Wall -fPIC 
-COMPILER_OPTS = -c -O2 -pipe -fPIC 
-LDFLAGS:=$(shell dpkg-buildflags --get LDFLAGS)
-CPPFLAGS:=$(shell dpkg-buildflags --get CPPFLAGS)
+COMPILER_OPTS = -c $(CXXFLAGS) -fPIC 
 
 #**************************************************************************
 LINKER = g++ 
diff --git a/cppForSwig/cryptopp/Makefile b/cppForSwig/cryptopp/Makefile
index 581d1aa..aecd444 100755
--- a/cppForSwig/cryptopp/Makefile
+++ b/cppForSwig/cryptopp/Makefile
@@ -1,4 +1,4 @@
-CXXFLAGS = -DNDEBUG -g -O2
+CXXFLAGS += -DNDEBUG
 #CXXFLAGS = -g
 # -fPIC is supported. Please report any breakage of -fPIC as a bug.
 CXXFLAGS += -fPIC
@@ -36,9 +36,6 @@ ISMINGW = $(shell $(CXX) --version 2>&1 | $(EGREP) -c "mingw")
 ifneq ($(GCC42_OR_LATER),0)
 ifeq ($(UNAME),Darwin)
 CXXFLAGS += -arch x86_64 -arch i386
-else
-#CXXFLAGS += -march=native
-CXXFLAGS += -mtune=generic
 endif
 endif
 
@@ -74,8 +71,6 @@ endif	# ISX86
 
 ifeq ($(UNAME),)	# for DJGPP, where uname doesn't exist
 CXXFLAGS += -mbnu210
-else
-CXXFLAGS += -pipe
 endif
 
 ifeq ($(UNAME),Linux)
