From 37e53dd3a8a73d07210673aedf455dc20dadbe36 Mon Sep 17 00:00:00 2001
From: Laurent Bachelier <laurent@bachelier.name>
Date: Mon, 28 Apr 2014 13:39:49 +0200
Subject: [PATCH] Remove Snappy

---
 cppForSwig/leveldb/build_detect_platform | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/cppForSwig/leveldb/build_detect_platform b/cppForSwig/leveldb/build_detect_platform
index 6e59c6f..35f5b87 100755
--- a/cppForSwig/leveldb/build_detect_platform
+++ b/cppForSwig/leveldb/build_detect_platform
@@ -183,17 +183,6 @@ EOF
         COMMON_FLAGS="$COMMON_FLAGS -DLEVELDB_PLATFORM_POSIX"
     fi
 
-    # Test whether Snappy library is installed
-    # http://code.google.com/p/snappy/
-    $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT 2>/dev/null  <<EOF
-      #include <snappy.h>
-      int main() {}
-EOF
-    if [ "$?" = 0 ]; then
-        COMMON_FLAGS="$COMMON_FLAGS -DSNAPPY"
-        PLATFORM_LIBS="$PLATFORM_LIBS -lsnappy"
-    fi
-
     # Test whether tcmalloc is available
     $CXX $CXXFLAGS -x c++ - -o $CXXOUTPUT -ltcmalloc 2>/dev/null  <<EOF
       int main() {}
-- 
1.8.3.2

