diff -Npur spacenavd-0.6.orig/Makefile.in spacenavd-0.6/Makefile.in
--- spacenavd-0.6.orig/Makefile.in	2017-01-14 17:01:33.194550674 +0100
+++ spacenavd-0.6/Makefile.in	2017-01-14 17:01:55.046552564 +0100
@@ -7,7 +7,7 @@ ctl = spnavd_ctl
 
 CC = gcc
 INSTALL = install
-CFLAGS = -pedantic -Wall $(dbg) $(opt) -fno-strict-aliasing -I$(srcdir)/src -I/usr/local/include $(add_cflags)
+CFLAGS = $(dbg) -fno-strict-aliasing -I$(srcdir)/src -I/usr/local/include $(add_cflags)
 LDFLAGS = -L/usr/local/lib $(xlib) $(add_ldflags)
 
 $(bin): $(obj)
