From: Nathan Phillip Brink <binki@gentoo.org>
Subject: Fix build order so that dynamic linking can happen properly.

--- a/live/Makefile.tail
+++ b/live/Makefile.tail
@@ -15,12 +15,10 @@
 MEDIA_SERVER_DIR = mediaServer
 MEDIA_SERVER_APP = $(MEDIA_SERVER_DIR)/mediaServer$(EXE)
 
-ALL =	$(LIVEMEDIA_LIB) \
-	$(GROUPSOCK_LIB) \
-	$(USAGE_ENVIRONMENT_LIB) \
+ALL =	$(USAGE_ENVIRONMENT_LIB) \
 	$(BASIC_USAGE_ENVIRONMENT_LIB) \
-	$(TESTPROGS_APP) \
-	$(MEDIA_SERVER_APP)
+	$(GROUPSOCK_LIB) \
+	$(LIVEMEDIA_LIB)
 all:	$(ALL)
 
 
