diff -Npur corewars-0.9.13.orig/configure.ac corewars-0.9.13/configure.ac
--- corewars-0.9.13.orig/configure.ac	2014-10-13 21:14:24.986941709 +0200
+++ corewars-0.9.13/configure.ac	2014-10-13 21:21:02.827976130 +0200
@@ -49,7 +49,7 @@ AM_PATH_GTK_2_0(2.0.0,,COREWARS=;AC_MSG_
    ************************************************************************
 ]))
 
-LIBS="$LIBS $GTK_LIBS"
+LIBS="$LIBS $GTK_LIBS -lm"
 dnl GTK_INCLUDE=`gtk-config --cflags`
 AC_SUBST(INCLUDES)
 INCLUDES="$INCLUDES $GTK_CFLAGS"
