diff -Npur corewars-0.9.13.orig/src/main-gui.c corewars-0.9.13/src/main-gui.c
--- corewars-0.9.13.orig/src/main-gui.c	2014-10-14 23:08:11.461410224 +0200
+++ corewars-0.9.13/src/main-gui.c	2014-10-14 23:09:38.931417792 +0200
@@ -15,6 +15,7 @@
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#define _GNU_SOURCE
 #include <stdlib.h>
 #include <gtk/gtk.h>
 #include <string.h>
