--- a/boinc-status	2020-04-30 22:57:38.099918744 +0200
+++ b/boinc-status	2020-04-30 22:57:13.059591555 +0200
@@ -11,7 +11,7 @@
 # Parameters to be set before using the script
 
 # The path where BOINC state files reside
-BOINC_PATH=/var/lib/boinc-client
+BOINC_PATH=/var/lib/boinc
 
 # The path where BOINC client executable resides (without the client name)
 # If client 'boinc' is in /usr/bin, you'll have BOINC_BIN_PATH=/usr/bin
@@ -39,7 +39,7 @@
 AUTHOR="Jean Philippe EIMER"
 EMAIL="phil@jpmr.org"
 
-BOINC_EXEC=boinc
+BOINC_EXEC=boinc_client
 
 # LOG file for BOINC, with its absolute path
 BOINC_LOG_FILE=$BOINC_PATH/stdoutdae.txt
@@ -47,8 +47,8 @@
 STATE_FILE=$BOINC_PATH/client_state.xml
 TASK_STATE_FILE=$BOINC_PATH/slots/SLOT/boinc_task_state.xml
 
-if [ -x /bin/pidof ]; then
-	PID_CMD=/bin/pidof
+if [ -x /usr/bin/pidof ]; then
+	PID_CMD=/usr/bin/pidof
 else
 	PID_CMD=/sbin/pidof
 fi
