From 79db1c5ae428ca05a0eb9a0485de02670f8f4b0b Mon Sep 17 00:00:00 2001
From: Maksym Sditanov <maxim.sditanov@lge.com>
Date: Thu, 13 Nov 2014 21:23:03 +0200
Subject: [PATCH] Add -fpermissive flag

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e3ba226..d12776b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,8 @@ pkg_check_modules(GLIB REQUIRED glib-2.0)
 include_directories(${GLIB_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${GLIB_CFLAGS_OTHER})
 
+webos_add_compiler_flags(ALL -fpermissive)
+
 function(nyx_get_pkgconfig_var varname nyx_pkgconfig_var usecache)
         execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=${nyx_pkgconfig_var} nyx
                         OUTPUT_VARIABLE pkgconfig_output
-- 
2.0.4

