diff --git a/libcpp/Makefile.in b/libcpp/Makefile.in
index 5fbba9b9c76..ebbca07f542 100644
--- a/libcpp/Makefile.in
+++ b/libcpp/Makefile.in
@@ -25,7 +25,7 @@ srcdir = @srcdir@
 top_builddir = .
 VPATH = @srcdir@
 INSTALL = @INSTALL@
-AR = @AR@
+AR = ar
 ARFLAGS = cru
 ACLOCAL = @ACLOCAL@
 AUTOCONF = @AUTOCONF@
diff --git a/libcpp/configure b/libcpp/configure
index 20051f88087..c7ea4d30193 100755
--- a/libcpp/configure
+++ b/libcpp/configure
@@ -658,7 +658,6 @@ ACLOCAL
 EGREP
 GREP
 CPP
-AR
 RANLIB
 ac_ct_CXX
 CXXFLAGS
diff --git a/libcpp/configure.ac b/libcpp/configure.ac
index 1efa96f7ca3..540efeb4629 100644
--- a/libcpp/configure.ac
+++ b/libcpp/configure.ac
@@ -12,7 +12,6 @@ AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_RANLIB
-AC_CHECK_TOOL(AR, ar)
 
 AC_USE_SYSTEM_EXTENSIONS
 AC_SYS_LARGEFILE
