--- ti99sim-0.13.0/rules.mak.a	2016-02-23 01:27:46.000000000 +0100
+++ ti99sim-0.13.0/rules.mak	2016-04-02 21:00:16.059317934 +0200
@@ -27,16 +27,14 @@
 DEBUG    := 1
 endif
 
-ifdef ARCH
-CFLAGS   += -march=$(ARCH)
-endif
-
 ifdef DEBUG
 CFLAGS   += -g3 -O0 -DDEBUG
 else
 CFLAGS   += -g3 -O3 -fomit-frame-pointer
 endif
 
+CFLAGS   += -DTI_DATA_DIR=\"$(DATA_DIR)\"
+
 ifdef LOGGER
 CFLAGS   += -D_REENTRANT -DLOGGER
 LIBS	 += -lrt
