--- ti99sim-0.15.0/rules.mak.a	2017-03-28 19:46:10.021143347 +0200
+++ ti99sim-0.15.0/rules.mak	2017-04-06 18:08:11.007403612 +0200
@@ -37,10 +37,6 @@
 DEBUG    := 1
 endif
 
-ifdef ARCH
-CFLAGS   += -march=$(ARCH)
-endif
-
 ifdef LTO
 CFLAGS   += -flto
 LFLAGS   += -flto
@@ -53,6 +49,8 @@
 LFLAGS   += -g3 -O3
 endif
 
+CFLAGS   += -DTI_DATA_DIR=\"$(DATA_DIR)\"
+
 ifdef LOGGER
 CFLAGS   += -D_REENTRANT -DLOGGER
 LIBS     += -lrt
