diff --git a/test/unittests/CMakeLists.txt b/test/unittests/CMakeLists.txt
index 359d619..4ca5908 100644
--- a/test/unittests/CMakeLists.txt
+++ b/test/unittests/CMakeLists.txt
@@ -4,7 +4,6 @@
 
 include(GoogleTest)
 
-hunter_add_package(GTest)
 find_package(GTest CONFIG REQUIRED)
 
 # Disable support for std::tr1::tuple in GTest. This causes problems in Visual Studio 2015.
diff --git a/tools/vmtester/CMakeLists.txt b/tools/vmtester/CMakeLists.txt
index a9eb0ab..d016d1d 100644
--- a/tools/vmtester/CMakeLists.txt
+++ b/tools/vmtester/CMakeLists.txt
@@ -4,7 +4,6 @@
 
 include(GNUInstallDirs)
 
-hunter_add_package(GTest)
 find_package(GTest CONFIG REQUIRED)
 
 # Disable support for std::tr1::tuple in GTest. This causes problems in Visual Studio 2015.
