--- a/configure.ac.orig	2019-07-27 00:11:30.905910532 +0200
+++ b/configure.ac	2019-07-27 00:23:31.490189965 +0200
@@ -62,8 +62,7 @@
 AM_CONDITIONAL([HAVE_GET_CURRENT_DIR_NAME], [test x$ac_cv_func_get_current_dir_name = xyes])
 AM_CONDITIONAL([HAVE_NANOSLEEP], [test x$ac_cv_func_nanosleep = xyes])
 
-AM_PATH_CPPUNIT(1.9.6,,[AC_MSG_ERROR([cppunit not found, tests disabled])])
-AM_CONDITIONAL([BUILD_TESTS], [test "x$no_cppunit" != "xyes"])
+AM_CONDITIONAL([BUILD_TESTS], [false])
 
 AC_CONFIG_COMMANDS_PRE([mkdir -p include/portable])
 
