--- openvpn-2.1_rc20~/easy-rsa/2.0/openssl-0.9.8.cnf~	2011-10-10 15:10:33.027120093 +0400
+++ openvpn-2.1_rc20~/easy-rsa/2.0/openssl-0.9.8.cnf	2011-10-10 15:11:07.447120924 +0400
@@ -283,8 +283,8 @@
 #pkcs11 = pkcs11_section
 
 [ pkcs11_section ]
-engine_id = pkcs11
-dynamic_path = /usr/lib/engines/engine_pkcs11.so
-MODULE_PATH = $ENV::PKCS11_MODULE_PATH
-PIN = $ENV::PKCS11_PIN
-init = 0
+#engine_id = pkcs11
+#dynamic_path = /usr/lib/engines/engine_pkcs11.so
+#MODULE_PATH = $ENV::PKCS11_MODULE_PATH
+#PIN = $ENV::PKCS11_PIN
+#init = 0
--- openvpn-2.1_rc20~/easy-rsa/2.0/openssl-1.0.0.cnf~	2011-10-10 15:10:36.257120163 +0400
+++ openvpn-2.1_rc20~/easy-rsa/2.0/openssl-1.0.0.cnf	2011-10-10 15:11:20.296121202 +0400
@@ -278,8 +278,8 @@
 #pkcs11 = pkcs11_section
 
 [ pkcs11_section ]
-engine_id = pkcs11
-dynamic_path = /usr/lib/engines/engine_pkcs11.so
-MODULE_PATH = $ENV::PKCS11_MODULE_PATH
-PIN = $ENV::PKCS11_PIN
-init = 0
+#engine_id = pkcs11
+#dynamic_path = /usr/lib/engines/engine_pkcs11.so
+#MODULE_PATH = $ENV::PKCS11_MODULE_PATH
+#PIN = $ENV::PKCS11_PIN
+#init = 0
