diff --git a/scripts/splitSources.py b/scripts/splitSources.py
index c14e5bf..2ac62df 100755
--- a/scripts/splitSources.py
+++ b/scripts/splitSources.py
@@ -48,7 +48,7 @@ def writeSourceToFile(lines):
 if __name__ == '__main__':
     filePath = sys.argv[1]
     # decide if file has multiple sources
-    lines = open(filePath, mode='rb', encoding='utf8').read().splitlines()
+    lines = open(filePath, mode='rb').read().splitlines()
     if lines[0][:12] == "==== Source:":
         hasMultipleSources = True
         writeSourceToFile(lines)
