diff --git a/libdevcore/CMakeLists.txt b/libdevcore/CMakeLists.txt
index 7b4342f..feb408b 100644
--- a/libdevcore/CMakeLists.txt
+++ b/libdevcore/CMakeLists.txt
@@ -34,6 +34,7 @@ set(sources
 )
 
 add_library(devcore ${sources})
+target_compile_features(devcore PUBLIC cxx_std_17)
 target_link_libraries(devcore PUBLIC jsoncpp Boost::boost Boost::filesystem Boost::system)
 target_include_directories(devcore PUBLIC "${CMAKE_SOURCE_DIR}")
 add_dependencies(devcore solidity_BuildInfo.h)
