diff --git a/libsolutil/CMakeLists.txt b/libsolutil/CMakeLists.txt
index 2c6b6b2..fb3d788 100644
--- a/libsolutil/CMakeLists.txt
+++ b/libsolutil/CMakeLists.txt
@@ -34,6 +34,7 @@ set(sources
 )
 
 add_library(solutil ${sources})
+target_compile_features(solutil PUBLIC cxx_std_17)
 target_link_libraries(solutil PUBLIC jsoncpp Boost::boost Boost::filesystem Boost::system)
 target_include_directories(solutil PUBLIC "${CMAKE_SOURCE_DIR}")
 add_dependencies(solutil solidity_BuildInfo.h)
