diff -ur Firebird-3.0.0.32483-0-a/builds/posix/make.defaults Firebird-3.0.0.32483-0-b/builds/posix/make.defaults
--- Firebird-3.0.0.32483-0-a/builds/posix/make.defaults	2016-04-14 16:07:29.000000000 +0200
+++ Firebird-3.0.0.32483-0-b/builds/posix/make.defaults	2016-04-21 17:49:17.504723178 +0200
@@ -154,7 +154,7 @@
 AR=			ar @AR_OPTIONS@
 LN=			@LN_S@
 RANLIB=			@RANLIB@
-BTYACC=$(ROOT)/extern/btyacc/btyacc
+BTYACC=/usr/bin/btyacc
 CLOOP=$(GEN_ROOT)/$(TARGET)/cloop/release/bin/cloop
 
 CC =	@CC@
diff -ur Firebird-3.0.0.32483-0-a/builds/posix/Makefile.in Firebird-3.0.0.32483-0-b/builds/posix/Makefile.in
--- Firebird-3.0.0.32483-0-a/builds/posix/Makefile.in	2016-04-14 16:07:29.000000000 +0200
+++ Firebird-3.0.0.32483-0-b/builds/posix/Makefile.in	2016-04-21 17:50:08.468844658 +0200
@@ -122,9 +122,7 @@
 # build external software
 #
 
-extern:
-	$(MAKE) -C $(ROOT)/extern/btyacc
-	$(MAKE) -C $(ROOT)/extern/cloop TARGET=release WITH_FPC=0 BUILD_DIR=$(TMP_ROOT)/cloop OUT_DIR=$(GEN_ROOT)/$(TARGET)/cloop
+extern: ;
 
 ifeq ($(STD_EDITLINE),false)
 ifeq ($(EDITLINE_FLG),Y)
