diff -ru tpmmanager-0.8.1.orig/src/TPMManager.cxx tpmmanager-0.8.1/src/TPMManager.cxx
--- tpmmanager-0.8.1.orig/src/TPMManager.cxx	2011-08-15 16:23:48.000000000 +0300
+++ tpmmanager-0.8.1/src/TPMManager.cxx	2012-02-19 14:38:58.000000000 +0200
@@ -409,7 +409,7 @@
 
 void TPM_Manager::slotProcessURL( const QString& url )
 {
-	QUrl myURL = QUrl::QUrl(url, QUrl::StrictMode);
+	QUrl myURL = QUrl(url, QUrl::StrictMode);
 	if ( !( QDesktopServices::openUrl(myURL) ) )
 		QMessageBox::critical( this, "Error: Opening a URL", "Could not open your favourite browser. " );
 }
