diff -ur SDLPoP-1.17.a/src/options.c SDLPoP-1.17.b/src/options.c
--- SDLPoP-1.17.a/src/options.c	2017-02-05 08:38:56.000000000 +0100
+++ SDLPoP-1.17.b/src/options.c	2017-12-15 17:54:12.466066642 +0100
@@ -207,7 +207,7 @@
 		process_word("pop_window_width", &pop_window_width, NULL);
 		process_word("pop_window_height", &pop_window_height, NULL);
 		process_boolean("use_correct_aspect_ratio", &use_correct_aspect_ratio);
-        process_boolean("enable_controller_rumble", &enable_controller_rumble);
+        enable_controller_rumble = 0;
         process_boolean("joystick_only_horizontal", &joystick_only_horizontal);
 
         if (strcasecmp(name, "levelset") == 0) {
diff -ur SDLPoP-1.17.a/src/seg009.c SDLPoP-1.17.b/src/seg009.c
--- SDLPoP-1.17.a/src/seg009.c	2017-02-05 08:38:56.000000000 +0100
+++ SDLPoP-1.17.b/src/seg009.c	2017-12-15 17:55:17.960412828 +0100
@@ -1933,7 +1933,7 @@
 // seg009:38ED
 void __pascal far set_gr_mode(byte grmode) {
 	if (SDL_Init(SDL_INIT_VIDEO | SDL_INIT_TIMER | SDL_INIT_NOPARACHUTE |
-				 SDL_INIT_GAMECONTROLLER | SDL_INIT_HAPTIC ) != 0) {
+				 SDL_INIT_GAMECONTROLLER ) != 0) {
 		sdlperror("SDL_Init");
 		quit(1);
 	}
