diff --git a/external/external.pro b/external/external.pro
index dd38186a..962ed07e 100644
--- a/external/external.pro
+++ b/external/external.pro
@@ -2,11 +2,12 @@ config += debug_and_release
 
 TEMPLATE      = subdirs
 
-SUBDIRS	= 	muparser_v225/src \
-			levmar-2.3/levmar-2.3.pro \
-			structuresynth-1.5/structuresynth.pro \
-			jhead-3.04/jhead-3.04.pro \
+#SUBDIRS	= 	muparser_v225/src \
+#			levmar-2.3/levmar-2.3.pro \
+#			structuresynth-1.5/structuresynth.pro \
+#			jhead-3.04/jhead-3.04.pro \
 #			openkinect/openkinect.pro
+SUBDIRS =              structuresynth-1.5/structuresynth.pro
 
 !CONFIG(system_bzip2)   SUBDIRS	+= 	bzip2-1.0.5/bzip2-1.0.5.pro
 !CONFIG(system_openctm) SUBDIRS	+= 	OpenCTM-1.0.3/openctm.pro
