diff --git a/lib/evmone/CMakeLists.txt b/lib/evmone/CMakeLists.txt
index 1fad575..f77879e 100644
--- a/lib/evmone/CMakeLists.txt
+++ b/lib/evmone/CMakeLists.txt
@@ -19,6 +19,7 @@ add_library(evmone
     limits.hpp
     opcodes_helpers.h
 )
+target_compile_features(evmone PUBLIC cxx_std_17)
 target_link_libraries(evmone PUBLIC evmc::evmc PRIVATE intx::intx ethash::keccak)
 target_include_directories(evmone PUBLIC
     $<BUILD_INTERFACE:${include_dir}>$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
diff --git a/test/utils/CMakeLists.txt b/test/utils/CMakeLists.txt
index 68e94fb..69e112b 100644
--- a/test/utils/CMakeLists.txt
+++ b/test/utils/CMakeLists.txt
@@ -10,6 +10,7 @@ add_library(testutils STATIC
     utils.hpp
 )
 
+target_compile_features(testutils PUBLIC cxx_std_17)
 target_link_libraries(testutils PRIVATE evmc::instructions)
 target_include_directories(testutils PUBLIC ${PROJECT_SOURCE_DIR})
 
